backend/src/Service/Admin/Order/AdminOrderService.php

```diff
@@ -842,5 +850,25 @@ public function filterOrdersNotAnsweredByTheStore(FilterOrdersPaidOrNotPaidByAdm
 
         return $response;
     }
-    
+
+    public function checkWhetherCaptainReceivedOrderForSpecificStore(int $captainProfileId, int $storeId, int|null $primaryOrderId): int
+    {
+        $orderEntity = $this->adminOrderManager->checkWhetherCaptainReceivedOrderForSpecificStore($captainProfileId, $storeId);
+       
+        if (!empty($orderEntity)) {
+            //if the order not main
+            if ($orderEntity->getOrderIsMain() !== true) {
+                return OrderResultConstant::CAPTAIN_RECEIVED_ORDER_FOR_THIS_STORE_INT;
+            }
+            //if the order main and (request order) related
+            if ($primaryOrderId === $orderEntity->getId()) {
+
+                return OrderResultConstant::CAPTAIN_NOT_RECEIVED_ORDER_FOR_THIS_STORE_INT;
```

Comment:
Why we prevent assigning the order to the captain while it is a sub order of an existing main one??

Reply:
I did not prevent the captain from doing so, but I did allow the captain to receive the sub-order, please check